openpmd_validator/check_h5.py

```diff
@@ -548,35 +550,46 @@ def check_meshes(f, iteration, v, extensionStates):
     # First element : number of errors
     # Second element : number of warnings
     result_array = np.array([ 0, 0]) 
-    
+
     # Find the path to the data
     base_path = "/data/%s/" % iteration
     valid, meshes_path = get_attr(f, "meshesPath")
-    if not valid :
-        print("Error: `meshesPath` is missing or malformed in '/'")
-        return( np.array([1, 0]) )
-    meshes_path = meshes_path.decode()
-
-    if os.path.join( base_path, meshes_path) != ( base_path + meshes_path ):
-        print("Error: `basePath`+`meshesPath` seems to be malformed "
-            "(is `basePath` absolute and ends on a `/` ?)")
-        return( np.array([1, 0]) )
+    if valid:
+        meshes_path = meshes_path.decode()
     else:
-        full_meshes_path = (base_path + meshes_path).encode('ascii')
-        # Find all the meshes
-        try:
-            list_meshes = list(f[full_meshes_path].keys())
-        except KeyError:
-            list_meshes = []
-    print( "Iteration %s : found %d meshes"
-        %( iteration, len(list_meshes) ) )
+        meshes_path = None
+        if v:
+            print("`meshesPath` attribute is missing in '/' "
+                  "(will not search for mesh records)")
+
+    if meshes_path:
+        if os.path.join( base_path, meshes_path) != ( base_path + meshes_path ):
+            print("Error: `basePath`+`meshesPath` seems to be malformed "
+                "(is `basePath` absolute and ends on a `/` ?)")
+            return( np.array([1, 0]) )
+        else:
+            full_meshes_path = (base_path + meshes_path).encode('ascii')
+            # if set, a directory must exist with this name
+            if not full_meshes_path in f:
+                print("Error: `basePath`+`meshesPath` are set but path '{0}' "
+                      "does not exist in file!".format(full_meshes_path))
+                return( np.array([1, 0]) )
+            # Find all the meshes
+            try:
+                list_meshes = list(f[full_meshes_path].keys())
+            except KeyError:
+                list_meshes = []
```

Comment:
Could you remove the `try / except KeyError` here? (Because you are basically testing for it above, with an `if / return` statement)?

Reply:
True, we are testing it properly now! :)